### ISSUE TITLE
Create first batch of recipes

### ISSUE THREAD

**Issue by Arshitha** [neuroimaging-cookbook/neuroimaging-cookbook.github.io#1]
This issue will document the list of ideas for recipes.

**Comment by Arshitha** [comment]
#### converting a dataset with dicom images to a BIDS formatted dataset using heudiconv

- **Background:**
Initially, I'd like to focus on using heudiconv on HPC (High Performance Computing) systems without admin access. Mostly because there is a lack of resources with a focus on HPC usage of heudiconv. Most institutions with private computational resources don't grant admin access to users to maintain privacy and a bunch of other security reasons. 

  Since docker containers hosted on docker hub for some of the tools like fmriprep, heudiconv, mriqc, etc can't be used directly. The workaround is to build a singularity container from the existing docker container and use that to for our analysis. 

  In the process, I'd also like to document other use cases for heudiconv as well. 

- **Method:**
Demonstrate the use of heudiconv with following types of datasets:
  - cross sectional datasets (each subject with a single session) with only structural scans
  - cross sectional datasets with both structural and functional scans
  - longitudinal datasets with multiple sessions but with only structural images
  - longitudinal datasets with multiple sessions with structural and functional scans 

- **Resources**
  - BIDS specifications: https://bids-specification.readthedocs.io/en/stable/
  - heudiconv: https://heudiconv.readthedocs.io/en/latest/
  - heudiconv tutorial with code snippets: https://slides.com/jameskent/deck-3 (this is where I got my idea from. we'd mostly be building on this)

- **Optional Resources** 
  - singularity: https://singularity.lbl.gov/

**Comment by bfuchs18** [comment]
- convert files into formats needed for imaging software (e.g., dcm to .nii)
- brain segmentation 
- preprocessing steps: skullstripping, motion correction, slice-time correction, smoothing, registration, normalization 
- creating timing files
- generate files, images, etc.? for quality checking functional data 
- analysis: model specification, generating contrast maps, creating GLM
- multiple-comparison correction (voxel-permutation testing, voxel-wise/clusterwise thresholding)
- extracting time-series data: from a mask, sphere, cluster

**Comment by bfuchs18** [comment]
non-neuroimaging specific code that might be involved in processing neuro data
- examples of ways to loop through participants, runs, etc.
- check to see if a script has been run for a particular subject/session, if yes skip subj, if no run script

**Comment by MartinezAddiego** [comment]
fMRI - RSA analysis, PCA Analysis, ICA analysis
EEG - preprocessing of EEG data: filtering, re-referencing, selecting/extracting epochs, baselining the current

**Comment by Arshitha** [comment]
I'm working on the recipe for heudiconv here https://hackmd.io/@b8Ow6HfjTyG4M3LRHcWP1A/rJz9oqM2w/edit (you can edit the document if you're signed in to hack.md) Feel free to add comments/edit/suggest edits

**Comment by shawnrhoads** [comment]
@bfuchs18 
- [x] dcm to nii (dcm2niix, dcm2niibatch), brik/head to nii (3dAFNItoNIFTI)
- [ ] 3dclustsim 
- [ ] brain segmentation

@Arshitha 
- [x] pydeface
- [ ] fMRIPrep
- [ ] MRIQC
- [x] bids-validator

@shawnrhoads 
- [ ] extract timeseries
- [ ] extract ROI from parcellation
- [ ] shuffle subject labels for permutation testing
- [ ] combine multiple 3D volume files to single 4D file
- [ ] trim first/last few TRs from 4D volume

**Comment by Arshitha** [comment]
Just adding a few more to the list that I'd like to work on:
- [ ] slicesdir to verify pydefce-ing results 
- [ ] OpenNeuro cli download 
- [ ] HeuDiConv on HPC
